File: 2021/python/01.py
ns = [int(l) for l in open("../../inputs/2021/01.txt").readlines()]

a = sum(1 if ns[i] < ns[i+1] else 0 for i in range(len(ns)-1))
b = sum(1 if ns[i] < ns[i+3] else 0 for i in range(len(ns)-3))

print(f"Part 1: {a}\nPart 2: {b}")

File: 2023/js/01.js
let input = await Bun.file("../../inputs/2023/01.txt").text();
input = input.trimEnd();

const solve = (data) => {
  data = data.split("\n");
  let sum = 0;
  data.forEach((l) => {
    let ds = [];
    [...l].forEach((c) => {
      if (c >= "0" && c <= "9") {
        ds.push(c);
      }
    });
    sum += parseInt(ds[0] + ds.splice(-1));
  });
  return sum;
};

console.log("Day 1: Trebuchet?!");
console.log("  Part 1:", solve(input));

[
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
].forEach(
  (v, i) => (input = input.replaceAll(v, v[0] + (i + 1) + v.slice(-1)))
);
console.log("  Part 2:", solve(input));
